### ISSUE TITLE
Failure in unmarshal surrogate unicode encoding (utf16) - BUG

### ISSUE THREAD

**Issue by fereidani** [mailru/easyjson#79]
Hi ,

Example JSON File is :
```
{
  "name":"Khashayar is not \ud83d\ude08"
}
```

And go model is : 
```
type Test struct {
	Name string `json:"name"`
}
```

Unmarshal test instance on mentioned JSON file will fail with this error:
parse error: invalid character 'K' after top-level value near offset 12 of 'Khashayar is not \ud83d\ude08"
}

as it's obvious json file is valid and represent :
```
{"name":"Khashayar is not 😈"}
```

the problem is from `processEscape` in `lexer.go` .
i will send pull request soon .

**Comment by stuartcarnie** [comment]
I have a fix for this – I'll send a pull request shortly

**Comment by GoWebProd** [comment]
Fixed in #84
